conn.go

```diff
@@ -14,6 +14,7 @@ import (
 type Conn struct {
 	stream inet.Stream
 	remote pstore.PeerInfo
+	relay  *Relay
```

Comment:
Unless we actually need the `Relay` here, let's replace this with `Host`.

P.S.: I would expect the Relay to have an accessor for the `peer.ID` of the relay, but it doesn't. That would've made the tagging code much clearer, as currently the reader needs to understand that the stream's remote ID is the relay, not the actual final peer.

Reply:
It made more sense to use the `Relay` object because it's the creator of those connections, but we can switch that to be a host.